code/datums/gamemodes/hunt.dm

```diff
@@ -0,0 +1,263 @@
+		/datum/job/terragov/command/pilot = 2,
+		/datum/job/terragov/engineering/chief = 1,
+		/datum/job/terragov/engineering/tech = 2,
+		/datum/job/terragov/requisitions/officer = 1,
+		/datum/job/terragov/medical/professor = 1,
+		/datum/job/terragov/medical/medicalofficer = 6,
+		/datum/job/terragov/medical/researcher = 2,
+		/datum/job/terragov/civilian/liaison = 1,
+		/datum/job/terragov/silicon/synthetic = 1,
+		/datum/job/terragov/silicon/ai = 1,
+		/datum/job/terragov/squad/engineer = 8,
+		/datum/job/terragov/squad/corpsman = 8,
+		/datum/job/terragov/squad/smartgunner = 1,
+		/datum/job/terragov/squad/leader = 1,
+		/datum/job/terragov/squad/standard = -1,
+		/datum/job/xenomorph = 2,
+		/datum/job/xenomorph/queen = 1
+	)
+
+
+/datum/game_mode/infestation/hunt/announce()
+	to_chat(world, "<span class='round_header'>The current map is - [SSmapping.configs[GROUND_MAP].map_name]!</span>")
+
+
+/datum/game_mode/infestation/hunt/can_start(bypass_checks = FALSE)
+	. = ..()
+	if(!.)
+		return
+	var/xeno_candidate = FALSE //Let's guarantee there's at least one xeno.
+	for(var/level = JOBS_PRIORITY_HIGH; level >= JOBS_PRIORITY_LOW; level--)
+		for(var/p in GLOB.ready_players)
+			var/mob/new_player/player = p
+			if(player.client.prefs.job_preferences[ROLE_XENO_QUEEN] == level && SSjob.AssignRole(player, SSjob.GetJobType(/datum/job/xenomorph/queen)))
+				xeno_candidate = TRUE
+				break
+			if(player.client.prefs.job_preferences[ROLE_XENOMORPH] == level && SSjob.AssignRole(player, SSjob.GetJobType(/datum/job/xenomorph)))
+				xeno_candidate = TRUE
+				break
+	if(!xeno_candidate && !bypass_checks)
+		to_chat(world, "<b>Unable to start [name].</b> No xeno candidate found.")
+		return FALSE
+
+
+/datum/game_mode/infestation/hunt/pre_setup()
+	. = ..()
+	addtimer(CALLBACK(SSticker.mode, .proc/map_announce), 5 SECONDS)
+
+
+/datum/game_mode/infestation/hunt/post_setup()
+	. = ..()
+	for(var/i in GLOB.xeno_resin_silo_turfs)
+		new /obj/structure/resin/silo(i)
+	scale_gear()
+	addtimer(CALLBACK(src, .proc/announce_bioscans, FALSE, 1), rand(30 SECONDS, 1 MINUTES)) //First scan shows no location but more precise numbers.
+
+/datum/game_mode/infestation/hunt/proc/map_announce()
+	if(!SSmapping.configs[GROUND_MAP].announce_text)
+		return
+
+	priority_announce(SSmapping.configs[GROUND_MAP].announce_text, SSmapping.configs[SHIP_MAP].map_name)
+
+
+/datum/game_mode/infestation/hunt/process()
+	if(round_finished)
+		return FALSE
+
+	//Automated bioscan / Queen Mother message
+	if(world.time > bioscan_current_interval)
+		announce_bioscans()
+		var/total[] = count_humans_and_xenos(count_flags = COUNT_IGNORE_XENO_SPECIAL_AREA)
+		var/marines = total[1]
+		var/xenos = total[2]
+		var/bioscan_scaling_factor = xenos / max(marines, 1)
+		bioscan_scaling_factor = max(bioscan_scaling_factor, 0.25)
+		bioscan_scaling_factor = min(bioscan_scaling_factor, 1.5)
+		bioscan_current_interval += bioscan_ongoing_interval * bioscan_scaling_factor
```

Comment:
Move this and all bioscan code to infestation typepath, the parent of this one and distress as both have the bioscan, and remove bioscan code from distress